src/psqlgraph/ext.py

```diff
@@ -1,59 +1,61 @@
-from collections import defaultdict
+"""A module for maintaining the extension of SQLAlchemy logic to define a graph ORM."""
 
+from __future__ import annotations
+
+import collections
+import functools
+from collections.abc import Iterable
+
+import sqlalchemy
+from sqlalchemy import event, orm, schema
 from sqlalchemy.ext import declarative
 
-from psqlgraph import Edge, Node
-from psqlgraph.base import CommonBase, LocalConcreteBase
-from psqlgraph.edge import AbstractEdge
-from psqlgraph.node import AbstractNode
+from psqlgraph import graph, voided
 
-BASE_CLASSES = defaultdict(dict)
-BASE_CLASSES[None] = {"node": Node, "edge": Edge}
+_GRAPHS: dict[str | None, graph.Graph] = {None: {"node": graph.Node, "edge": graph.Edge}}
+_ORM_BASES: dict[str | None, type] = collections.defaultdict(declarative.declarative_base)
+_ORM_BASES[None] = graph.Base
 
-ORM_BASES = defaultdict(lambda: declarative.declarative_base(cls=CommonBase))
+# Add the listener for configuring the all graphs before the mapper is configured for
+# them. This generally happens upon the first use of any of the defined entities
+event.listen(
+    orm.mapper, "before_configured", functools.partial(graph.configure_graph, _GRAPHS.values())
+)
 
 
-def get_orm_base(package_namespace):
+def get_orm_base(package_namespace: str | None) -> type:
     """Helper function to get the appropriate sqlalchemy base class
     Args:
        package_namespace (str): module namespace
     """
-    return ORM_BASES[package_namespace]
-
-
-def get_abstract_edge(package_namespace=None):
-    return BASE_CLASSES[package_namespace]["edge"]
+    return _ORM_BASES[package_namespace]
 
 
-def get_abstract_node(package_namespace=None):
-    return BASE_CLASSES[package_namespace]["node"]
+def get_abstract_edge(package_namespace: str | None = None) -> type[graph.AbstractEdge]:
+    return _GRAPHS[package_namespace]["edge"]
 
 
-def get_class_prefix(pkg_namespace):
-    """Return an abstract class name prefix for the provided package
-    Args:
-        pkg_namespace (str): valid package name e.g gpas, bio
-    Returns:
-        str: class name prefix eg. AbstractGpas
-    """
-    return f"{pkg_namespace.title()}"
+def get_abstract_node(package_namespace: str | None = None) -> type[graph.AbstractNode]:
+    return _GRAPHS[package_namespace]["node"]
 
 
-def create_base_class(pkg_namespace, is_node=True):
+def create_base_class(package_namespace: str) -> graph.Graph:
     """Dynamically creates an abstract base class that extends either the Node or Edge class
     Args:
         pkg_namespace (str): package namespace
         is_node (bool): if True creates a node abstract class else creates an edge
     Returns:
         class: A dynamically generated abstract class
     """
+    base = get_orm_base(package_namespace)
+    edge, node = graph.__bind_orm__(base)
```

Comment:
I am unsure about creating base classes using the `__bind_orm__` technique. The core issue this was solving is the isolation of sqlalchemy bases at the class name. This allowed for separating say an AlignedReads loaded in gdcdictionary vs the same AlignedReads loaded from biodictionary. I see what the __bind_orm__ is doing, but I am skeptical that is sufficient, because what you end up with as the class name will be something like this `psqlgraph.graph.ext.__bind_orm__.<locals>.Node` even with different package namespaces.

Reply:
I've done some deep diving on this and I can't create this issue. They do appear to have to the same qualified name but they are not actually the same class and have distinct MROs.